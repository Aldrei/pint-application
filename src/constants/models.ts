export const NeighborhoodModel = {
  pt: {
    name: 'Bairro'
  }
};
