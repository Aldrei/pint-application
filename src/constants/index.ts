export const API = {
  AUTH: 'oauth/access_token'
}