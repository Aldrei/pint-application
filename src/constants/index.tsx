import DashboardOutlinedIcon from '@mui/icons-material/DashboardOutlined';
import MapsHomeWorkIcon from '@mui/icons-material/MapsHomeWork';
import ContactsIcon from '@mui/icons-material/Contacts';

import { ROUTES } from './routes';

export const API = {
  AUTH: 'oauth/access_token',
  REVOKE: 'api/oauth/revoke',
  PROPERTIES: {
    LIST: 'api/properties',
    SHOW: (code: string) => `api/properties/${code}`,
    STORE: 'api/properties',
    UPDATE: (id: string) => `api/properties/${id}`,
    SEARCH: (search: string) => `api/properties/search/${search}`,
    VIDEOS: {
      LIST: (code: string) => `api/properties/${code}/videos?page=1`,
      UPLOAD: (code: string) => `api/properties/${code}/videos`,
      DELETE: (id: string, video_id: string) => `api/properties/${id}/videos/${video_id}`,
    },
    PHOTOS: {
      LIST: (code: string) => `api/properties/${code}/photos?page=1`,
      STORE: (code: string) => `api/properties/${code}/photos`,
      UPDATE: (id: string, photo_id: string) => `api/properties/${id}/photos/${photo_id}`,
      UPDATE_POSITIONS: (code: string) => `api/properties/${code}/photos/all/update-posicoes`,
      DELETE: (id: string, photo_id: string) => `api/properties/${id}/photos/${photo_id}`,
    }
  },
  OWNERS: {
    SEARCH: (search: string) => `api/owners/search/${search}`
  },
  EMPLOYEES: {
    SEARCH: (search: string) => `api/employees/search/${search}`
  },
  CITIES: {
    SEARCH: (search: string) => `api/cities/search/${search}`
  },
  NEIGHBORHOODS: {
    SEARCH: (search: string, cityId: string) => `api/cities/${cityId}/neighborhoods/search/${search}`
  },
};

export const MENU = {
  PART_ONE: {
    VISAO_GERAL: {
      desc: 'Visão Geral',
      icon: <DashboardOutlinedIcon />,
      route: ROUTES.dashboard.go()
    },
    IMOVEIS: {
      desc: 'Imóveis',
      icon: <MapsHomeWorkIcon />,
      route: ROUTES.propertiesList.go({})
    },
    CLIENTES: {
      desc: 'Clientes',
      icon: <ContactsIcon />,
      route: ''
    },
    EQUIPE: {
      desc: 'Equipe',
      icon: '',
      route: ''
    },
    BAIRROS: {
      desc: 'Bairro',
      icon: '',
      route: ''
    },
  },
  PART_TWO: {
    SLIDE_DO_SITE: {
      desc: 'Slide do Site',
      icon: '',
      route: ''
    },
    MENSAGENS_DO_SITE: {
      desc: 'Mensagens do Site',
      icon: '',
      route: ''
    },
    SOLICITACOES_DO_SITE: {
      desc: 'Solicitações do Site',
      icon: '',
      route: ''
    },
    AGENCIAMENTOS_DO_SITE: {
      desc: 'Agenciamentos do Site',
      icon: '',
      route: ''
    },
  }
};

export const MAX_PHOTOS_BY_PROPERTY = 50;
