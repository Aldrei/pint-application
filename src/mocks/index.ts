export const PROPERTIES_LIST = {
  'paginate':{
    'data':[
      {
        'id':50,
        'client_id':1,
        'city_id':4303905,
        'neighborhood_id':2,
        'owner_id':46,
        'agent_id':1,
        'broker_id':1,
        'code':50,
        'codeTipo':null,
        'codePretty':null,
        'nomeImovel':'',
        'matricula':'',
        'lote':'',
        'quadra':'',
        'empreendimento':0,
        'placa':0,
        'possuiFoto':1,
        'exclusividade':0,
        'exclusividadePeriodoInicio':'',
        'exclusividadePeriodoFim':'',
        'finalidade':'',
        'categoria':'Residencial',
        'tipo':'Apartamento',
        'status':'Vendendo',
        'aluguelPeriodoInicio':'',
        'aluguelPeriodoFim':'',
        'dormitorio':'2',
        'garagem':'1',
        'lavanderia':1,
        'telhado':'',
        'forro':'',
        'piso':'',
        'aberturas':'',
        'alarme':1,
        'portaoEletronico':1,
        'pocoArtesiano':0,
        'cercaEletrica':0,
        'cameraDeVideo':0,
        'nascerDoSol':'',
        'descGeral':'02 dorm, sala,garagem.',
        'dataAgenciamento':'',
        'dataVenda':'',
        'apNomeCondominio':'',
        'apPredio':'',
        'apPavimento':'',
        'apTotalPavimentos':'',
        'apApto':'',
        'apElevador':1,
        'areaTotal':'',
        'areaConstruida':'',
        'areaFrente':'',
        'areaFundos':'',
        'areaDireita':'',
        'areaEsquerda':'',
        'valor':'',
        'valorCondicaoDeComissao':'',
        'valorPorcentagemDeComissao':'0.00',
        'valorPorcentagemDoCorretor':'0.00',
        'valorPorcentagemDoAgenciador':'0.00',
        'valorCondominio':'',
        'valorIPTUPago':'',
        'valorIPTU':'',
        'valorInssPago':'',
        'valorInss':'0.00',
        'valorIndiceDoCUB':'0.00',
        'valorAlgoParaRegularizar':'',
        'valorAlgoParaRegularizarDesc':'',
        'condAgente':'',
        'condPrestacao':'',
        'condSaldoDevedor':'',
        'condPrazo':'',
        'condReajuste':'',
        'condFGTS':0,
        'condDoacao':0,
        'condFinanciamento':0,
        'condCartaConsorcio':0,
        'condPartePermuta':0,
        'condSituacao':'',
        'condObs':'',
        'localEstado':'RS',
        'localIdCidade':'1',
        'localIdBairro':null,
        'localLogradouro':'',
        'localNumero':'',
        'localCEP':'',
        'localImediacoes':'',
        'latitude':null,
        'longitude':null,
        'zoom':null,
        'sitePublicarImovel':1,
        'sitePublicarValor':1,
        'siteImovelDestaque':0,
        'siteAcesso':'689',
        'sitePublicarMapa':null,
        'videoURL':'',
        'idUltimaTransacao':'0',
        'created_at':'2018-02-18 13:56:50',
        'updated_at':'2019-08-02 20:40:40',
        'desc':'Apartamento, 2 dormit\u00f3rios, garagem 1 carro',
        'title':'Apartamento, 2 dorm.',
        'hasExclusividade':false,
        'site_url':'http://csimovel.com/imovel/50/apartamento/campo-bom',
        'city':{
          'data':{
            'id':4303905,
            'name':'Campo Bom',
            'state':{
              'id':43,
              'name':'Rio Grande do Sul',
              'abbr':'RS',
              'created_at':'2018-02-18 16:56:49',
              'updated_at':'2018-02-18 16:56:49'
            },
            'long_desc':'Campo Bom, RS'
          }
        },
        'neighborhood':{
          'data':{
            'id':2,
            'client_id':1,
            'city_id':4303905,
            'idBairro':'',
            'nome':'Centro',
            'idCidade':'1',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:37',
            'city':{
              'data':{
                'id':4303905,
                'name':'Campo Bom',
                'state':{
                  'id':43,
                  'name':'Rio Grande do Sul',
                  'abbr':'RS',
                  'created_at':'2018-02-18 16:56:49',
                  'updated_at':'2018-02-18 16:56:49'
                },
                'long_desc':'Campo Bom, RS'
              }
            }
          }
        },
        'owner':{
          'data':{
            'id':46,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'idProprietario':'',
            'nomeRazao':'Olegario',
            'cpfCnpj':'',
            'rg':'',
            'cnh':'',
            'pessoa':'Fisica',
            'inscricaoEstadual':'',
            'sexo':'Masculino',
            'estadoCivil':'Casado(a)',
            'dataNascimento':'',
            'naturalidade':'',
            'profissao':'',
            'renda':'',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'fixo':'',
            'celular':'',
            'fax':'',
            'email':'',
            'email2':'',
            'obs':'',
            'conjNome':'',
            'conjCpf':'',
            'conjRg':'',
            'conjCnh':'',
            'conjDataNascimento':'0000-00-00',
            'conjNaturalidade':'',
            'conjProfissao':'',
            'conjRenda':'0.00',
            'conjPai':'',
            'conjMae':'',
            'conjMesmoEndereco':'o',
            'conjEstado':'',
            'conjIdCidade':'0',
            'conjIdBairro':'0',
            'conjLogradouro':'',
            'conjNumero':'',
            'conjCep':'',
            'conjApto':'',
            'conjFixo':'',
            'conjCelular':'',
            'conjFax':'',
            'conjEmail':'',
            'conjEmail2':'',
            'conjSpc':'',
            'conjSpcEntrada':'0000-00-00',
            'conjSpcSaida':'0000-00-00',
            'conjSpcValor':'0.00',
            'conjObs':'',
            'foto':'',
            'fotoMini':'',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:33'
          }
        },
        'broker':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'agent':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'photo':{
          'data':{
            'id':465,
            'property_id':50,
            'name':'5bf2583ec2633e60ecdb69ed247bc5ef.jpg',
            'ordem':1,
            'rotate':null,
            'thumb':'https://imobmobile.com.br/photos/thumb/5bf2583ec2633e60ecdb69ed247bc5ef.jpg',
            'normal':'https://imobmobile.com.br/photos/normal/5bf2583ec2633e60ecdb69ed247bc5ef.jpg'
          }
        }
      },
      {
        'id':56,
        'client_id':1,
        'city_id':4303905,
        'neighborhood_id':13,
        'owner_id':49,
        'agent_id':1,
        'broker_id':1,
        'code':56,
        'codeTipo':63,
        'codePretty':'CAA063',
        'nomeImovel':'',
        'matricula':'',
        'lote':'',
        'quadra':'',
        'empreendimento':0,
        'placa':0,
        'possuiFoto':1,
        'exclusividade':0,
        'exclusividadePeriodoInicio':'',
        'exclusividadePeriodoFim':'',
        'finalidade':'',
        'categoria':'Residencial',
        'tipo':'Casa Alvenaria',
        'status':'Vendendo',
        'aluguelPeriodoInicio':'',
        'aluguelPeriodoFim':'',
        'dormitorio':'3',
        'garagem':'3',
        'lavanderia':1,
        'telhado':'',
        'forro':'',
        'piso':'',
        'aberturas':'',
        'alarme':1,
        'portaoEletronico':1,
        'pocoArtesiano':0,
        'cercaEletrica':1,
        'cameraDeVideo':1,
        'nascerDoSol':'',
        'descGeral':'03 dorm (1 su\u00edte c/hidro e closet ),cozinha sob medida,pain\u00e9is,porcelanato,gesso,\u00e1gua quente,lareira,piscina,persianas automatizadas.',
        'dataAgenciamento':'',
        'dataVenda':'',
        'apNomeCondominio':'',
        'apPredio':'',
        'apPavimento':'',
        'apTotalPavimentos':'',
        'apApto':'',
        'apElevador':0,
        'areaTotal':'300.00',
        'areaConstruida':'229.00',
        'areaFrente':'12.00',
        'areaFundos':'12.00',
        'areaDireita':'25.00',
        'areaEsquerda':'25.00',
        'valor':'850000.00',
        'valorCondicaoDeComissao':'',
        'valorPorcentagemDeComissao':'0.00',
        'valorPorcentagemDoCorretor':'0.00',
        'valorPorcentagemDoAgenciador':'0.00',
        'valorCondominio':'',
        'valorIPTUPago':'',
        'valorIPTU':'',
        'valorInssPago':'',
        'valorInss':'0.00',
        'valorIndiceDoCUB':'0.00',
        'valorAlgoParaRegularizar':'',
        'valorAlgoParaRegularizarDesc':'',
        'condAgente':'',
        'condPrestacao':'',
        'condSaldoDevedor':'',
        'condPrazo':'',
        'condReajuste':'',
        'condFGTS':0,
        'condDoacao':0,
        'condFinanciamento':0,
        'condCartaConsorcio':0,
        'condPartePermuta':0,
        'condSituacao':'',
        'condObs':'',
        'localEstado':'RS',
        'localIdCidade':'1',
        'localIdBairro':null,
        'localLogradouro':'Rodrigo Jos\u00e9 Piltz',
        'localNumero':'83',
        'localCEP':'',
        'localImediacoes':'',
        'latitude':null,
        'longitude':null,
        'zoom':null,
        'sitePublicarImovel':0,
        'sitePublicarValor':0,
        'siteImovelDestaque':0,
        'siteAcesso':'1704',
        'sitePublicarMapa':null,
        'videoURL':'',
        'idUltimaTransacao':'0',
        'created_at':'2018-02-18 13:56:50',
        'updated_at':'2021-08-11 14:04:05',
        'valorPub':'850.000,00',
        'desc':'Casa Alvenaria, 3 dormit\u00f3rios, garagem 3 carros',
        'title':'Casa Alvenaria, 3 dorm.',
        'hasExclusividade':false,
        'site_url':'http://csimovel.com/imovel/56/casa-alvenaria/campo-bom',
        'city':{
          'data':{
            'id':4303905,
            'name':'Campo Bom',
            'state':{
              'id':43,
              'name':'Rio Grande do Sul',
              'abbr':'RS',
              'created_at':'2018-02-18 16:56:49',
              'updated_at':'2018-02-18 16:56:49'
            },
            'long_desc':'Campo Bom, RS'
          }
        },
        'neighborhood':{
          'data':{
            'id':13,
            'client_id':1,
            'city_id':4303905,
            'idBairro':'',
            'nome':'Solar do Campo',
            'idCidade':'1',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:37',
            'city':{
              'data':{
                'id':4303905,
                'name':'Campo Bom',
                'state':{
                  'id':43,
                  'name':'Rio Grande do Sul',
                  'abbr':'RS',
                  'created_at':'2018-02-18 16:56:49',
                  'updated_at':'2018-02-18 16:56:49'
                },
                'long_desc':'Campo Bom, RS'
              }
            }
          }
        },
        'owner':{
          'data':{
            'id':49,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'idProprietario':'',
            'nomeRazao':'Eliezer',
            'cpfCnpj':'',
            'rg':'',
            'cnh':'',
            'pessoa':'Fisica',
            'inscricaoEstadual':'',
            'sexo':'Masculino',
            'estadoCivil':'Casado(a)',
            'dataNascimento':'',
            'naturalidade':'',
            'profissao':'',
            'renda':'',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'fixo':'',
            'celular':'(51) 8236-4900',
            'fax':'',
            'email':'',
            'email2':'',
            'obs':'',
            'conjNome':'',
            'conjCpf':'',
            'conjRg':'',
            'conjCnh':'',
            'conjDataNascimento':'0000-00-00',
            'conjNaturalidade':'',
            'conjProfissao':'',
            'conjRenda':'0.00',
            'conjPai':'',
            'conjMae':'',
            'conjMesmoEndereco':'1',
            'conjEstado':'',
            'conjIdCidade':'0',
            'conjIdBairro':'0',
            'conjLogradouro':'',
            'conjNumero':'',
            'conjCep':'',
            'conjApto':'',
            'conjFixo':'(51) 9388-0596',
            'conjCelular':'(51) 9236-4900',
            'conjFax':'',
            'conjEmail':'',
            'conjEmail2':'',
            'conjSpc':'',
            'conjSpcEntrada':'0000-00-00',
            'conjSpcSaida':'0000-00-00',
            'conjSpcValor':'0.00',
            'conjObs':'',
            'foto':'',
            'fotoMini':'',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:33'
          }
        },
        'broker':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'agent':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'photo':{
          'data':{
            'id':491,
            'property_id':56,
            'name':'271ccaf01e123f96c9174dde8a65af5d.jpg',
            'ordem':1,
            'rotate':null,
            'thumb':'https://imobmobile.com.br/photos/thumb/271ccaf01e123f96c9174dde8a65af5d.jpg',
            'normal':'https://imobmobile.com.br/photos/normal/271ccaf01e123f96c9174dde8a65af5d.jpg'
          }
        }
      },
      {
        'id':63,
        'client_id':1,
        'city_id':4303905,
        'neighborhood_id':6,
        'owner_id':56,
        'agent_id':1,
        'broker_id':1,
        'code': null,
        'codeTipo':null,
        'codePretty':null,
        'nomeImovel':'',
        'matricula':'',
        'lote':'',
        'quadra':'',
        'empreendimento':0,
        'placa':1,
        'possuiFoto':1,
        'exclusividade':0,
        'exclusividadePeriodoInicio':'',
        'exclusividadePeriodoFim':'',
        'finalidade':'',
        'categoria':'Residencial',
        'tipo':'Terreno',
        'status': '',
        'aluguelPeriodoInicio':'',
        'aluguelPeriodoFim':'',
        'dormitorio':'0',
        'garagem':'0',
        'lavanderia':0,
        'telhado':'',
        'forro':'',
        'piso':'',
        'aberturas':'',
        'alarme':0,
        'portaoEletronico':0,
        'pocoArtesiano':0,
        'cercaEletrica':0,
        'cameraDeVideo':0,
        'nascerDoSol':'',
        'descGeral':'Terreno com linda vista.',
        'dataAgenciamento':'',
        'dataVenda':'',
        'apNomeCondominio':'',
        'apPredio':'',
        'apPavimento':'',
        'apTotalPavimentos':'',
        'apApto':'',
        'apElevador':0,
        'areaTotal':'310.15',
        'areaConstruida':'',
        'areaFrente':'10.65',
        'areaFundos':'10.74',
        'areaDireita':'29.00',
        'areaEsquerda':'29.00',
        'valor':'84800.00',
        'valorCondicaoDeComissao':'',
        'valorPorcentagemDeComissao':'0.00',
        'valorPorcentagemDoCorretor':'0.00',
        'valorPorcentagemDoAgenciador':'0.00',
        'valorCondominio':'',
        'valorIPTUPago':'',
        'valorIPTU':'',
        'valorInssPago':'',
        'valorInss':'0.00',
        'valorIndiceDoCUB':'0.00',
        'valorAlgoParaRegularizar':'',
        'valorAlgoParaRegularizarDesc':'',
        'condAgente':'',
        'condPrestacao':'',
        'condSaldoDevedor':'',
        'condPrazo':'',
        'condReajuste':'',
        'condFGTS':0,
        'condDoacao':0,
        'condFinanciamento':0,
        'condCartaConsorcio':0,
        'condPartePermuta':0,
        'condSituacao':'',
        'condObs':'',
        'localEstado':'RS',
        'localIdCidade':'1',
        'localIdBairro':null,
        'localLogradouro':'Waldemar Mossmann',
        'localNumero':'',
        'localCEP':'',
        'localImediacoes':'',
        'latitude':null,
        'longitude':null,
        'zoom':null,
        'sitePublicarImovel':0,
        'sitePublicarValor':0,
        'siteImovelDestaque':1,
        'siteAcesso':'34',
        'sitePublicarMapa':null,
        'videoURL':'',
        'idUltimaTransacao':'0',
        'created_at':'2018-02-18 13:56:50',
        'updated_at':'2020-02-25 18:23:49',
        'valorPub':'84.800,00',
        'desc':'Terreno',
        'title':'Terreno',
        'hasExclusividade':false,
        'site_url':'http://csimovel.com/imovel/63/terreno/campo-bom',
        'city':{
          'data':{
            'id':4303905,
            'name':'Campo Bom',
            'state':{
              'id':43,
              'name':'Rio Grande do Sul',
              'abbr':'RS',
              'created_at':'2018-02-18 16:56:49',
              'updated_at':'2018-02-18 16:56:49'
            },
            'long_desc':'Campo Bom, RS'
          }
        },
        'neighborhood':{
          'data':{
            'id':6,
            'client_id':1,
            'city_id':4303905,
            'idBairro':'',
            'nome':'Morada da Brisa',
            'idCidade':'1',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:37',
            'city':{
              'data':{
                'id':4303905,
                'name':'Campo Bom',
                'state':{
                  'id':43,
                  'name':'Rio Grande do Sul',
                  'abbr':'RS',
                  'created_at':'2018-02-18 16:56:49',
                  'updated_at':'2018-02-18 16:56:49'
                },
                'long_desc':'Campo Bom, RS'
              }
            }
          }
        },
        'owner':{
          'data':{
            'id':56,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'idProprietario':'',
            'nomeRazao':'Luiz Meister',
            'cpfCnpj':'',
            'rg':'',
            'cnh':'',
            'pessoa':'Fisica',
            'inscricaoEstadual':'',
            'sexo':'Masculino',
            'estadoCivil':'Casado(a)',
            'dataNascimento':'',
            'naturalidade':'',
            'profissao':'',
            'renda':'',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'fixo':'',
            'celular':'(51) 9288-5199',
            'fax':'',
            'email':'',
            'email2':'',
            'obs':'',
            'conjNome':'',
            'conjCpf':'',
            'conjRg':'',
            'conjCnh':'',
            'conjDataNascimento':'0000-00-00',
            'conjNaturalidade':'',
            'conjProfissao':'',
            'conjRenda':'0.00',
            'conjPai':'',
            'conjMae':'',
            'conjMesmoEndereco':'1',
            'conjEstado':'',
            'conjIdCidade':'0',
            'conjIdBairro':'0',
            'conjLogradouro':'',
            'conjNumero':'',
            'conjCep':'',
            'conjApto':'',
            'conjFixo':'',
            'conjCelular':'',
            'conjFax':'',
            'conjEmail':'',
            'conjEmail2':'',
            'conjSpc':'',
            'conjSpcEntrada':'0000-00-00',
            'conjSpcSaida':'0000-00-00',
            'conjSpcValor':'0.00',
            'conjObs':'',
            'foto':'',
            'fotoMini':'',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2018-02-18 15:33:33'
          }
        },
        'broker':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'agent':{
          'data':{
            'id':1,
            'user_id':null,
            'client_id':1,
            'city_id':null,
            'neighborhood_id':null,
            'nome':'Mara Mello',
            'dataNascimento':'0000-00-00',
            'cargo':'4',
            'salarioBase':'0.00',
            'creci':'',
            'usuarioDoSistema':'1',
            'ativo':'1',
            'estado':'',
            'idCidade':'0',
            'idBairro':'0',
            'logradouro':'',
            'numero':'',
            'cep':'',
            'apto':'',
            'email':'mara-mello@hotmail.com',
            'email2':'',
            'celular':'(51) 99989-4181',
            'fixo':'',
            'foto':'',
            'fotoMini':'',
            'publicarNoSite':'0',
            'oculto':'0',
            'created_at':'2018-02-18 13:56:50',
            'updated_at':'2020-11-30 14:09:35',
            'emails':{
              'data':[
                       
              ]
            }
          }
        },
        'photo': null
      },
    ],
    'meta':{
      'pagination':{
        'total':272,
        'count':10,
        'per_page':10,
        'current_page':2,
        'total_pages':28,
        'links':{
          'previous':'https://imobmobile.com.br/api/properties?page=1',
          'next':'https://imobmobile.com.br/api/properties?page=3'
        }
      }
    }
  }
};
