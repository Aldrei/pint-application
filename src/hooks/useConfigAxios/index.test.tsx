import { AxiosResponse, AxiosPromise, AxiosRequestConfig } from 'axios';
import React from 'react';

import { clearAccessTokenReducer } from '../../reducers/auty';

import { AXIOS_RESPONSE_SUCCESS, AXIOS_RESPONSE_ERROR, AXIOS_CONFIG_REQUEST_SUCCESS_DEFAULT, AXIOS_CONFIG_REQUEST_SUCCESS_CUSTOM, PERSIST_IMOB_AUTH, PERSIST_IMOB_AUTH_WITH_WRONG_DATA } from '../../mocks/constants';

import { useConfigAxios, interceptorRequestSuccess, interceptorResponseSuccess, interceptorResponseError } from './index';
import { render } from '@testing-library/react';
import api from './instance';

jest.mock('./instance');

jest.mock('../../stores/hooks', () => ({
  useAppDispatch: jest.fn(),
}));

jest.mock('../../reducers/auty', () => ({
  clearAccessTokenReducer: jest.fn(),
}));

const ComponentTest = (): React.ReactElement => {
  useConfigAxios();
  return <span>Test</span>;
};

describe('useConfigAxios', () => {
  const clearAccessTokenReducerMocked = clearAccessTokenReducer as jest.MockedFunction<typeof clearAccessTokenReducer>;
  const apiMocked = api as jest.MockedFunction<typeof api>;

  it('Should render correctly', () => {
    const axiosCreateMocked: AxiosPromise = jest.fn(() => ({
      interceptors: jest.fn()
    })) as unknown as AxiosPromise;
    apiMocked.mockReturnValue(axiosCreateMocked);

    const nodeEl = render(<ComponentTest />);
    expect(nodeEl.baseElement).toMatchSnapshot();
    expect(nodeEl.baseElement).toHaveTextContent('Test');
  });

  it('Should intercept success request and set access token', () => {
    window.localStorage.setItem('persist:imob', JSON.stringify(PERSIST_IMOB_AUTH));

    const dataRequestDefault = AXIOS_CONFIG_REQUEST_SUCCESS_DEFAULT as unknown as AxiosRequestConfig;
    const dataRequestCustom = AXIOS_CONFIG_REQUEST_SUCCESS_CUSTOM as unknown as AxiosRequestConfig;
    
    expect(interceptorRequestSuccess(dataRequestDefault)).toStrictEqual(dataRequestCustom);
  });

  it('Should intercept success request with empty access token', () => {
    window.localStorage.setItem('persist:imob', '');

    const dataRequestDefault = AXIOS_CONFIG_REQUEST_SUCCESS_DEFAULT as unknown as AxiosRequestConfig;
    
    expect(interceptorRequestSuccess(dataRequestDefault)).toStrictEqual(dataRequestDefault);
  });

  it('Should intercept success request with reject', () => {
    window.localStorage.setItem('persist:imob', JSON.stringify(PERSIST_IMOB_AUTH_WITH_WRONG_DATA));

    const dataRequestDefault = AXIOS_CONFIG_REQUEST_SUCCESS_DEFAULT as unknown as AxiosRequestConfig;

    try {
      expect(interceptorRequestSuccess(dataRequestDefault));
    } catch(e) {      
      expect(e).toEqual({
        error: 'error_to_get_access_token',
      });
    }
  });

  it('Should intercept success response', () => {
    const axiosResponse = AXIOS_RESPONSE_SUCCESS as unknown as AxiosResponse;
    expect(interceptorResponseSuccess(axiosResponse)).toStrictEqual(JSON.parse(JSON.stringify(axiosResponse)));
  });

  it('Should intercept error response', () => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const dataResponse = AXIOS_RESPONSE_ERROR as any;
    expect(interceptorResponseError(dataResponse, jest.fn())).toStrictEqual(JSON.parse(JSON.stringify(dataResponse.response)));
    expect(clearAccessTokenReducerMocked.mock.calls.length).toBe(1);
  });
});
