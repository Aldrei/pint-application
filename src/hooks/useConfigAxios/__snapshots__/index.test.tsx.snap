// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useConfigAxios Should render correctly 1`] = `
<body>
  <div>
    <span>
      Test
    </span>
  </div>
</body>
`;
